spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BatchNormalization.scala

```diff
@@ -57,7 +57,7 @@ class BatchNormalization[T: ClassTag](
   private val initBias: Tensor[T] = null,
   private val initGradWeight: Tensor[T] = null,
   private val initGradBias: Tensor[T] = null
-)(implicit ev: TensorNumeric[T]) extends TensorModule[T] with Initializable {
+)(implicit ev: TensorNumeric[T]) extends TensorModule[T] with Initializable with MklInt8Convertible {
```

Comment:
please sync in dnn layer

Reply:
MklInt8Convertible should only be applicable to MKLDNN layers, no?

Reply:
@jason-dai We only save one version of model which is based on the orignal BLAS backend. MKL-DNN layer is not visible to user. and can't use `saveModule` also. With @zhangxiaoli73 work, we can convert the original BLAS model at runtime based on the `engineType` property. 

But Int8 in MKL-DNN needs a model specific infomation (scales). They should be saved somewhere. This trait as a bridge between the two version of layers is a solution.  Then bigdl can save the scales in the model (protobuf).  and will sync them to dnn models when user call `quantize` and the engine is mkldn. 

Of cause, the initial version, only `Convolution` and `Linear` need the scales. But because of the optimization, the scales should be updated based on the fusion rules, such as conv + bn fusion will update conv's output scales to bn's output scales.

Reply:
@jason-dai One driving motivation is we plan to decouple layer implementation from underlying engine and make layer unified, this is planned along with mkl-dnn upgrade soon. Also it should be better to keep format only and runtime engine irrelevant layer definition. That's why we gave up the option to save DNN defined models